test/generate-json.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+
+import argparse
+import json
+import time
+import os
+import struct
+
+# PingMessage is generated by this project
+import sys
+sys.path.append("../src/protocol/python")
+import PingMessage as Message
+
+# argparse for future devices
+parser = argparse.ArgumentParser(description="Script to generate json test description file for ping protocol!")
+parser.add_argument('--device', action="store", type=str, default="Ping1D", help="Ping device type.")
+args = parser.parse_args()
+
+# Generate fake data to decode
+def _generateFakeData(format):
+    return bytearray(os.urandom(struct.calcsize(format)))
+
+# Create payload
+def _values(format, fakedata, payload_fields):
+    values = list(struct.unpack(format, fakedata))
+    data = {}
+    for a,b in zip(values, payload_fields):
+        data[b] = a
+    return data
+
+#Checksum = sum(0 -> n) & 0xffff
+def _buildChecksum(h, p):
+    hUnpacked = struct.unpack("<BBBBBBBB", h)
+    if p is None:
+        p = ()
+    pFormat = '<' + (len(p) * 'B')
+    try:
+        pUnpacked = struct.unpack(pFormat, p)
+    except:
+        pUnpacked = ()
+
+    hSize = len(h)
+    pSize = len(p)
+    sumOfBytes = 0
+    for i in range(0, hSize):
+        sumOfBytes += hUnpacked[i]
+    for i in range(0, pSize):
+        sumOfBytes += pUnpacked[i]
+
+    checksum = sumOfBytes & 0xffff
+    return checksum
+
+# Pack message
+def _packRaw(id, m_message, m_payload, m_destination):
+    #Pack payload first, because metadata is required for the header
+    finalPayload = m_payload
+    #Needed to build header
+    payloadLength = struct.calcsize(m_message['format'])
+
+    #Create and pack header
+    header = [b'B', b'R', payloadLength, id, 0, m_destination]
+    finalHeader = struct.pack('<ccHHBB', *header)
+
+    #Create Checksum
+    checksum = _buildChecksum(finalHeader, finalPayload)
+    finalChecksum = struct.pack('<H', checksum)
+    return b''.join([finalHeader, finalPayload, finalChecksum])
+
+# Create json file with everything that we need
```

Comment:
I suggest adding a `if __name__ == "__main__":` here, so this part of the code doesn't run if someone tries to import this module from somewhere else .